### PR TITLE
Ds photogram industrial 2

### PR DESCRIPTION
**Pull request by dsimp**

### CODE REVIEW

**Review by Testell — APPROVED** [review]
Heavy

File: app/controllers/photos_controller.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    # Use callbacks to share common setup or constraints between actions.
+    def set_photo
+      @photo = Photo.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def photo_params
+      params.require(:photo).permit(:image, :comments_count, :likes_count, :caption, :owner_id)
+    end
+end
```
Comment (JoseMDO) [inline app/controllers/photos_controller.rb:70]: Your controllers all look great!

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
+  has_many :own_photos, class_name: "Photo", foreign_key:"owner_id"
+  has_many :comments, foreign_key: "author_id"
+  has_many :sent_follow_requests, foreign_key: :sender_id, class_name: "FollowRequest"
+  has_many :received_follow_requests, foreign_key: :recipient_id, class_name: "FollowRequest"
+  has_many :likes, foreign_key: :fan_id
+  has_many :own_photos, foreign_key: :owner_id, class_name: "Photo"
```
Comment (JoseMDO) [inline app/models/user.rb:38]: Your associations are all great, good job Djvon!

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,9 @@
 Rails.application.routes.draw do
+  resources :likes
+  resources :follow_requests
+  resources :comments
+  resources :photos
+  devise_for :users
```
Comment (JoseMDO) [inline config/routes.rb:6]: Routes all look good!

**Review by JoseMDO — APPROVED** [review]
Great job using scaffolding to create all your standard RCAV files for your app Djvon!

**Pull request merged**